lua/telescope/_extensions/project/main.lua

```diff
@@ -70,7 +72,11 @@ M.project = function(opts)
       map('i', '<c-w>', _actions.change_workspace)
 
       local on_project_selected = function()
-        _actions.find_project_files(prompt_bufnr, hidden_files)
+        if browse_by_default then
+          _actions.browse_project_files(prompt_bufnr, hidden_files)
+        else
+          _actions.find_project_files(prompt_bufnr, hidden_files)
+        end
```

Comment:
Great addition!

A suggestion -
You can extend it allow other actions as default

- search inside files within your project
- change to the selected project's directory without opening it
- find a recently opened file within your project